clippy_lints/src/stable_sort_primitive.rs

```diff
@@ -112,8 +112,8 @@ impl LateLintPass<'_> for StableSortPrimitive {
                 STABLE_SORT_PRIMITIVE,
                 expr.span,
                 format!(
-                    "used {} instead of {} to sort primitive type `{}`",
-                    detection.method.unstable_name(),
+                    "used `{}` on primitive type `{}`. An unstable sort would perform faster \
```

Comment:
I would use `span_lint_and_then` here and split up the "``used `sort`...``" message and the explaining message:

```rust
span_lint_and_then(
    cx,
    STABLE_SORT_PRIMITIVE,
    expr.span,
    format!(
        "used `{}` on primitive type `{}`",
        detection.method.stable_name(),
        detection.slice_type,
    ),
    |diag| {
        diag.span_suggestion(
            expr.span,
            "try",
            format!(
                "{}.{}({})",
                detection.slice_name,
                detection.method.unstable_name(),
                detection.method_args,
            ),
            Applicability::MachineApplicable,
        );
        diag.note("an unstable sort would perform faster without any observable difference for this data type");
    }
)
```

Reply:
I have updated the code using `span_lint_and_then`.